### PR TITLE
Establish Docker Image build provenance and move to artifact registry

### PR DESCRIPTION
**Pull request by drewroengoogle**
Bug: b/242225018

Creates build provenance for the following docker images:
* Docker image utilized by the flutter dashboard
* Flutter dashboard
* Flutter autosubmit

This required a couple of significant changes:
* Docker images are now deployed to artifact registry instead of container registry.
  * This change was necessary because provenance is able to be generated in artifact registry easily, but not in container registry.
* Originally,`gcloud app deploy` automatically generated the docker image and uploaded it.
  * In order to generate provenance, we needed to create the images ourselves. Therefore, the build and deploy have been split into different cloud builds, with a main cloud build configuration executing the build and deploy cloudbuild configs in order.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/cloudbuild_app_dart_docker.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Provide instructions for google Cloud Build to build and upload the flutter
+# dashboard docker image to artifact registry.
+
+steps:
+  # Build dashboard.
+  - name: us-docker.pkg.dev/$PROJECT_ID/flutter/flutter
+    entrypoint: '/bin/bash'
+    args: ['cloud_build/dashboard_build.sh']
```
Comment (keyonghan) [inline app_dart/cloudbuild_app_dart_docker.yaml:8]: Let's move this step to `cloudbuild_app_dart_deploy.yaml`, as this belongs to the deploy process.

**Review by sealesj — APPROVED** [review]

File: app_dart/cloudbuild_app_dart_docker.yaml (hunk 1 above)
Comment (drewroengoogle) [reply app_dart/cloudbuild_app_dart_docker.yaml:8]: @keyonghan I might have misunderstood, but I thought this step was required for setting up the docker image that is deployed in the next step?

Comment (keyonghan) [reply app_dart/cloudbuild_app_dart_docker.yaml:8]: If you take a look at `cloud_build/dashboard_build.sh`, you can see this step builds the `dashboard` and copies files over which prepares the app_dart deploy.

Comment (drewroengoogle) [reply app_dart/cloudbuild_app_dart_docker.yaml:8]: @keyonghan So I moved this step into the deployment and it looks like index.html can't be found: https://version-test-dot-flutter-dashboard.appspot.com/

**Review by keyonghan — APPROVED** [review]
LGTM!

File: app_dart/cloudbuild_app_dart_docker.yaml (hunk 1 above)
Comment (drewroengoogle) [reply app_dart/cloudbuild_app_dart_docker.yaml]: Discussed a bit more with Keyong - added some comments to increase clarity. Resolving!

**Pull request merged**